packages/pocket-ic/src/nonblocking.rs

```diff
@@ -78,6 +78,8 @@ pub struct PocketIc {
     http_gateway: Option<HttpGatewayInfo>,
     server_url: Url,
     reqwest_client: reqwest::Client,
+    // do not delete the instance when dropping this handle if this handle was created for an existing instance
```

Comment:
Is the idea that dfx holds on to the only keep=false struct instance, and every library-create instance is formed from the remote info and gets keep=true, and so will not send an instance deletion signal to the server on being dropped? 

Have you considered making it refcounted instead?

Reply:
The thing is that with this, we have to trust the dfx implementation to behave. Perhaps with refcounting we can do without that assumption.

Reply:
> Is the idea that dfx holds on to the only keep=false struct instance, and every library-create instance is formed from the remote info and gets keep=true, and so will not send an instance deletion signal to the server on being dropped?

Correct.

> Have you considered making it refcounted instead?

Where would you count? In a file on disk that will be updated atomically?

Reply:
The server instance could know how many library struct instances have registered with it and delete itself. That means that instances start out with count 1 when created via ::new(), and get bumped when somebody calls from_existing_instance()

Reply:
> The server instance could know how many library struct instances have registered with it.

That would require an additional endpoint on the server.

Reply:
> a docstring might suffice to stop users from using it wrong

gave it a try: https://github.com/dfinity/ic/pull/2721/commits/62b338557ee84c1478cc6559e33d6b3844516cdf